Add read/write action unit test to transaction obj

howjmay opened the pull request:
<<Describe the changes>>
#772 
# Test Plan:
<<Outline how the reviewer can verify & test your changes here>>

howjmay committed:
Add read/write action unit test

thibault-martinez commented on common/model/tests/test_transaction.c:
```diff
@@ -49,12 +49,94 @@ void test_reset(void) {
+  iota_transaction_t *transaction = transaction_new();
+
+  // write transaction values
+  transaction_set_address(transaction, TEST_ADDRESS_0);
```
These functions needs `flex_trit_t *`, not `tryte_t *`

howjmay replied:
oh my fault

thibault-martinez commented on common/model/tests/test_transaction.c:
```diff
@@ -49,12 +49,94 @@ void test_reset(void) {
+
+  // read transaction values
+  TEST_ASSERT_EQUAL_MEMORY(transaction_address(transaction), TEST_ADDRESS_0,
+                           NUM_TRYTES_ADDRESS);
```
This will crash if trit encoding is not set to 3, you need to use the size of the field

thibault-martinez commented on common/model/tests/test_transaction.c:
```diff
@@ -49,12 +49,94 @@ void test_reset(void) {
+  uint64_t test_snapshot_index = 2345;
+  bool test_solid = true;
+  uint64_t test_arrival_timestamp = 1477037812737;
+  iota_transaction_t *transaction = transaction_new();
```
No need to use dynamic allocation here

howjmay committed:
Fix errors

fix wrong function parameter from `tryte_t *` to `flex_trit_t *`

fix wrong TEST_ASSERT size parameter

remove dynamic allocation

howjmay committed:
chnage test function name

thibault-martinez committed:
Merge branch 'develop' into trans_test

thibault-martinez commented on common/model/tests/test_transaction.c:
```diff
@@ -49,12 +49,125 @@ void test_reset(void) {
+  flex_trit_t test_branch[FLEX_TRIT_SIZE_243];
+  uint64_t test_attachment_timestamp = 1477037821737;
+  uint64_t test_attachment_timestamp_lower = 1477037831737;
+  uint64_t test_attachment_timestamp_upper = 1477037841737;
```
Can you also put all these test values in `common/model/tests/defs.h` ? They could be useful elsewhere.

thibault-martinez commented on common/model/tests/test_transaction.c:
```diff
@@ -49,12 +49,125 @@ void test_reset(void) {
+  iota_transaction_t transaction;
+
+  // write transaction values
+  flex_trits_from_trytes(test_address, FLEX_TRIT_SIZE_243, TEST_ADDRESS_0,
```
flex trits functions always use size in trits or in trytes, not in flex trits. Please refer to the prototype doccumentation. Here you need to replace `FLEX_TRIT_SIZE_243` with `NUM_TRITS_ADDRESS` and so on...

thibault-martinez commented on common/model/tests/test_transaction.c:
```diff
@@ -49,12 +49,125 @@ void test_reset(void) {
+
+  // write transaction values
+  flex_trits_from_trytes(test_address, FLEX_TRIT_SIZE_243, TEST_ADDRESS_0,
+                         NUM_TRYTES_ADDRESS, sizeof(TEST_ADDRESS_0));
```
Also please replace the sizeofs by the actual size, usually the same as the previous parameter.

thibault-martinez commented on common/model/tests/test_transaction.c:
```diff
@@ -49,12 +49,125 @@ void test_reset(void) {
+
+  // read transaction values
+  TEST_ASSERT_EQUAL_MEMORY(transaction_address(&transaction), test_address,
+                           sizeof(test_address));
```
It would be awesome if you could test that the masks associated with each field are set

howjmay committed:
fix wrong arguments error

howjmay committed:
Add field masks tests

test transaction_set_signature and transaction_set_message
respectively.

howjmay committed:
Merge branch 'develop' into trans_test

thibault-martinez commented on common/model/transaction.h:
```diff
@@ -18,6 +18,7 @@ extern "C" {
 
 #define NUM_TRITS_SERIALIZED_TRANSACTION 8019
 #define NUM_TRITS_SIGNATURE 6561
+#define NUM_TRITS_MESSAGE 6561
```
👍

thibault-martinez committed:
Merge branch 'develop' into trans_test

thibault-martinez committed:
model/transaction: reset fields before testing

thibault-martinez commented:
Thanks !
